src/System.IO.Abstractions.TestingHelpers/MockFileSystem.cs

```diff
@@ -396,6 +403,14 @@ private bool DirectoryExistsWithoutFixingPath(string path)
             }
         }
 
+        private bool FileIsReadOnly(string path)
+        {
+            lock (files)
```

Comment:
Can we get of the `lock` here? Looks unnecessary to me because we already lock at the call site.